### DISCUSSION TITLE
Search Results and Search Filters Performance

### DISCUSSION THREAD

**Discussion by kalrav12learning** [microsoft-search/pnp-modern-search#2987] [Q&A]
Hi Team, 

We display 350000 items in search results through PnP Search and it takes longer time to load .

In this scenario, is there any configuration setting or customization way to improve performance to load search results and filters? 

Thanks In Advance

**Comment by kasperbolarsen** [comment]
A slight rewrite of a well known meme: show me the query !!!

**Reply by wobba** [reply]
..and how large is each page of results? Sharing the scenario and setup helps when answering to ensure we can provide guidance.

**Reply by kalrav12learning** [reply]
Thanks wobba for reply. 

We have migrated content from SharePoint 2013 to SharePoint Online and these contents in GB in order to we display minimum  350000  amd maximium 750000 records in each page.

**Reply by kasperbolarsen** [reply]
And there are no significant properties which might serve as a candidate for search verticals?
Like is it a RFP, a contract or any other document type. 
Which division/department owns the document,  or which financial year does it belongs to?

**Reply by wobba** [reply]
> Thanks wobba for reply.
> 
> We have migrated content from SharePoint 2013 to SharePoint Online and these contents in GB in order to we display minimum 350000 amd maximium 750000 records in each page.

Max search result page size is 500 items. And displaying more than maybe 50 per page, unless you have a grid view does not make sense. The number of items you index and search across does not matter.. it's how many is returned on a single query.

**Reply by kalrav12learning** [reply]
Hi wobba, 

Sorry , let me correct here . 

We display 10 items per page as per paging  and we dispaly  350000 results in one tab and 750000 results in another tab . 

We also display refinement filters . Search results and refinement filters both takes longer time to load . is there any way to improve performance in these two widgets ?

**Reply by wobba** [reply]
Impossible to give a good answer unfortunately. What kind of queries are you passing down? How are you sorting/ranking data? It's not about the full data-set, it's about how many items match any given query. What kind of KQL filters are you using?

If you look at the network tab, is it an API issue on loading slowly, or rendering slowly once data is returned?
